pyproject.toml

```diff
@@ -45,12 +67,11 @@ package-mode = false
 
 [dependency-groups]
 dev = [
-  "black==24.8.0",
   "sqlfluff==3.2.0",
-  "ruff>=0.8.2",
+  "ruff==0.12.9",
```

Comment:
Is there something preventing us from doing `ruff>=0.12.9` here?

Reply:
We pin version numbers in the monorepo to improve reproducibility and to provide greater transparency regarding the versions installed. I also find it simpler to always pin version numbers rather than deciding case by case which strategy to use in different scenarios. This approach has worked well for our projects in TypeScript and Java.

In Python, however, there are additional constraints that can affect dependency resolution, so a strict pinning strategy may not be as effective or straightforward. We can address this challenge when we reach that stage.

I would also welcome a side-by-side comparison of the two strategies for Python, strict pinning vs. more flexible version ranges, to serve as a reference for making an informed decision.

Reply:
I think for dev dependencies (things like ruff, black, mypy etc.) this is fine. But when the project is a package(Schematic, SynapsePythonClient) that others use, we want to be more flexible and provide the widest range possible.